src/main/java/concatenatedwordschallenge/processing/InitialWordsProcessor.java

```diff
@@ -0,0 +1,36 @@
+package concatenatedwordschallenge.processing;
+
+import concatenatedwordschallenge.util.TaskFileReader;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class InitialWordsProcessor {
+
+    private List<String> words;
+
+    private Map<Integer, List<String>> wordsWithLengths;
```

Comment:
let's make this class [stateless](https://stackoverflow.com/questions/9735601/what-is-stateless-object-in-java)

Reply:
Could You please explain, is it enough to consider class stateless if I make fields final and dont write any methods which change fields after creation? 
Or maybe all getters should also return a new copy of requested object? (for me sounds as immutable already)